packages/vite/src/node/plugins/css.ts

```diff
@@ -1101,10 +1101,20 @@ UrlRewritePostcssPlugin.postcss = true
 
 function rewriteCssUrls(
   css: string,
-  replacer: CssUrlReplacer
+  replacer: CssUrlReplacer,
+  file?: string
 ): Promise<string> {
   return asyncReplace(css, cssUrlRE, async (match) => {
     const [matched, rawUrl] = match
+    const inLess = file?.endsWith('.less')
+    const inSass = file?.endsWith('.sass')
+    const inScss = file?.endsWith('.scss')
+    if (
+      (inLess && rawUrl.startsWith('@')) ||
+      ((inSass || inScss) && rawUrl.startsWith('$'))
+    ) {
+      return `url('${rawUrl}')`
```

Comment:
```suggestion
      return `url(${rawUrl})`
```
`rawUrl` include quotes if needed.